### DISCUSSION TITLE
Are there situations where it's impossible to fee-bump a force-closure following an SCB?

### DISCUSSION THREAD

**Discussion by MegalithicBTC** [lightningnetwork/lnd#9057] [Q&A]
I'm doing some tests regarding the SCB feature ( https://docs.lightning.engineering/lightning-network-tools/lnd/disaster-recovery ) and I think I may have discovered a repeatable situation where it's not possible to fee-bump the force-close transactions.

I've been able to get myself into this state 3 times in a row, on TestNet.  Here is my test scenario which reliably puts me in this situation. 

Node A:  Channel opener, SCB initiator
Node B: Receives channel open from Node A, and is requested to force-close during the SCB process.

1) Node A opens a 100,000 sats channel to Node B. ......
2) Node A goes through a SCB process  
3) SCB is successful, and Node B force-closes the channel.
4) On Node B, the channel now appears in `waiting_close_channels`, like this: 

```
"waiting_close_channels": [
        {
            "channel": {
                "remote_node_pub": "02f69a0dcac367c025aa9b6d530983bad014c81c3f8f6439c329f457cb2d967dba",
                "channel_point": "a96c4c7b30694c0866e988c94817854f0fda527bb06a8684939da0c9cbec5211:1",
                "capacity": "100000",
                "local_balance": "0",
                "remote_balance": "96226",
                "local_chan_reserve_sat": "1000",
                "remote_chan_reserve_sat": "1000",
                "initiator": "INITIATOR_REMOTE",
                "commitment_type": "ANCHORS",
                "num_forwarding_packages": "0",
                "chan_status_flags": "ChanStatusBorked|ChanStatusCommitBroadcasted|ChanStatusLocalCloseInitiator",
                "private": true,
                "memo": "",
                "custom_channel_data": ""
            },
            "limbo_balance": "0",
            "commitments": {
                "local_txid": "0b6870f175584a6ff273b5a5abfa3a0651644da7cfed4da14a1bc4ac8378019b",
                "remote_txid": "df92a09dff655cf55aa6429be32375005162cadad10964a75edc0aae9a48bf3c",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "3114",
                "remote_commit_fee_sat": "3114",
                "remote_pending_commit_fee_sat": "0"
            },
            "closing_txid": "0b6870f175584a6ff273b5a5abfa3a0651644da7cfed4da14a1bc4ac8378019b",
            "closing_tx_hex": ""
        },
```
 ..... back on Node A, I don't get the close txid, I only see this data in `waiting_close_channels`:

```
"waiting_close_channels":  [
        {
            "channel":  {
                "remote_node_pub":  "020c3b59b1854d31b94c0dd9df1019d251b4cd5565129f7ad16e3ec8c4671afd8c",
                "channel_point":  "a96c4c7b30694c0866e988c94817854f0fda527bb06a8684939da0c9cbec5211:1",
                "capacity":  "100000",
                "local_balance":  "0",
                "remote_balance":  "0",
                "local_chan_reserve_sat":  "0",
                "remote_chan_reserve_sat":  "0",
                "initiator":  "INITIATOR_LOCAL",
                "commitment_type":  "ANCHORS",
                "num_forwarding_packages":  "0",
                "chan_status_flags":  "ChanStatusRestored",
                "private":  true,
                "memo":  ""
            },
            "limbo_balance":  "0",
            "commitments":  {
                "local_txid":  "",
                "remote_txid":  "",
                "remote_pending_txid":  "",
                "local_commit_fee_sat":  "0",
                "remote_commit_fee_sat":  "0",
                "remote_pending_commit_fee_sat":  "0"
            },
            "closing_txid":  "",
            "closing_tx_hex":  ""
        }
    ]
```
.... so, the force-close has been initiated, which is good, BUT, I think due to the fact that there is no balance on Node B's side, combined with the current high fees on TestNet, the force-close transaction was put in at a very low fee rate for the current (testnet) mempool: https://mempool.space/testnet/tx/0b6870f175584a6ff273b5a5abfa3a0651644da7cfed4da14a1bc4ac8378019b ....

 it was put in at 17 sat per Vb........ 

5) So I could wait here, but what if I had an impatient user who needs his BTC back ASAP? How could I fee-bump this for the user? 

6) I can't fee-bump it on Node B, I think, because the balance was all on Node A's side! No outputs are going to Node B! ...... and Node A doesn't know about the transaction (I think this is how SCBs work, right?) .. 

7) I have tried the command `lncli wallet  bumpclosefee  --conf_target 12 0b6870f175584a6ff273b5a5abfa3a0651644da7cfed4da14a1bc4ac8378019b:0` on BOTH nodes, and tried both output index `0` and `1`, but on BOTH nodes I get the error `[lncli] channel not found`

... so is this just the expected (and imperfect) behavior of an SCB, that sometimes it's not possible to fee bump?

Or am I just not using the right command?

thanks!

**Comment by MegalithicBTC** [comment]
Oh, I see the problem with my `bumpclosefee`, I need to use the channel point instead of close txid, let me try that now...

**Comment by MegalithicBTC** [comment]
I think I need to actually run `bumpforceclosefee` .. and I think these docs might need updating as they don't include `bumpforceclosefee` https://docs.lightning.engineering/lightning-network-tools/lnd/unconfirmed-bitcoin-transactions

**Comment by MegalithicBTC** [comment]
AND - I think I need to run `lnd v0.18.3-beta.rc1` to get the fix for this issue https://github.com/lightningnetwork/lnd/issues/8837

**Reply by guggero** [reply]
I think what you're running into is https://github.com/lightningnetwork/lnd/issues/8280.
We need to fix this in `lnd` of course, but until then there's [`chantools pullanchor`](https://github.com/lightninglabs/chantools/blob/master/doc/chantools_pullanchor.md).

**Comment by MegalithicBTC** [comment]
Ah, so I DID run into a known issue. Thanks.
